src/preprocessor.md

```diff
@@ -31,7 +31,52 @@ There also exists a subset of defines called Macros. These are in the format: `#
 
 ## #if / #elif / #else / #ifdef / #endif
 
-TODO
+These conditional defines work very similarly to the regular `if()` and `else` logic that we saw earlier, except that you have to close the code block with `#endif`. The code within the true statements gets compiled, while the others do not compile. Like defines, they get sort of 'swapped out' before the code is run, so they're not typically used with variables, but with other defines.
```

Comment:
> not typically used with other variables

more like *can't* be used...

>While the others do not compile 

Yes they do, otherwise your code wouldn't work. The contents *aren't included*. And what are "the others"?